tokio/src/sync/mod.rs

```diff
@@ -431,6 +431,25 @@
 //!   number of permits, which tasks may request in order to enter a critical
 //!   section. Semaphores are useful for implementing limiting or bounding of
 //!   any kind.
+//!
+//! # Runtime compatibility
+//!
+//! All synchronization primitives provided in this module are runtime agnostic.
+//! You can freely move them between different instances of the Tokio runtime
+//! or even use them from non-Tokio runtimes.
+//!
+//! The only exception to this is the [`mpsc::Sender::send_timeout`] method.
+//! It relies on functionality from the [`time`](crate::time) module which
+//! does not allow you to call it from non-Tokio runtimes.
+//! However, [`Senders`](mpsc::Sender) can still be moved between different
+//! instances of the Tokio runtime without you losing the ability to call
+//! `send_timeout` on them.
+//!
+//! When used in a Tokio runtime, the synchronization primitives in this module
+//! participate in [cooperative scheduling](crate::task#cooperative-scheduling),
+//! periodically yielding back control to the runtime to avoid starving it.
+//! This is Tokio specific behavior that does not come into effect when the
+//! synchronization primitive is used in a non-Tokio runtime.
```

Comment:
I would like this sentence to be more direct about the fact that, when you are not in a Tokio runtime, cooperative scheduling doesn't do anything, and that the synchroniation primitives still work just fine.